### ISSUE TITLE
Please make console output more verbose

### ISSUE THREAD

**Issue by remal** [arturbosch/detekt#1695]
I'm using Detekt with Gradle. Detekt Gradle plugin generates beautiful HTML reports, but I don't use them. I'm looking at console output instead. Please make console output more verbose.

For example, I'm getting this message:
```
ReturnCount - [...] at ...
```

Message like this would be much more useful:
```
ReturnCount - [...] at ...
    Restrict the number of return statements in methods.
    Function ... has ... return statements which exceeds the limit of 2.
```

**Comment by schalkms** [comment]
Thanks for your feedback!
I think we decided to not clutter the console output with too much information.
For this purpose we have several report formats.
Therefore, I'm against making this change.

**Comment by remal** [comment]
I'm not using report files at all, as I'm using GitLab CI. So I need to print as much useful information to then console as possible. Maybe it makes sense to do it configurable...

**Comment by schalkms** [comment]
But you can still persist files (in this case detekt's report files) even when using GitLab CI.

**Comment by remal** [comment]
I can persist. Sure. But having verbose input whould be more convinient.

**Comment by schalkms** [comment]
We will see what other users think about your suggestion.
Therefore, I put the according label on this issue.

**Comment by remal** [comment]
BTW is there a programming API that will allows me to build my own Gradle plugin with verbose output? Could you please suggest a place where to read about it? If the API exists, how stable is it?

**Comment by schalkms** [comment]
@remal that's certainly possible with detekt. If you want to go that way, you can do that with custom reports.
An example is located [here](https://arturbosch.github.io/detekt/extensions.html#custom-reports).
However, it's not a Gradle plugin but rather a way of extending detekt's report functionality.
If you need help just ask here or in detekt's Slack channel.

**Comment by arturbosch** [comment]
@remal we would highly appreciate a more verbose `ConsoleReport` (https://github.com/arturbosch/detekt/blob/ed23c35505f382390ff1c57c855a33763723125e/detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ConsoleReport.kt#L14-L23)

Ours is here: https://github.com/arturbosch/detekt/blob/ed23c35505f382390ff1c57c855a33763723125e/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/console/FindingsReport.kt#L10

One verbose which is off by default and that could be turned on in the config file would be awesome (https://github.com/arturbosch/detekt/blob/ed23c35505f382390ff1c57c855a33763723125e/detekt-cli/src/main/resources/default-detekt-config.yml#L18-L25).

**Comment by schalkms** [comment]
closed because of the arguments made in PR #1704
